src/strategies/evolution-land-dao/index.ts

```diff
@@ -0,0 +1,130 @@
+import { formatUnits } from '@ethersproject/units';
+import { getProvider, call } from '../../utils';
+
+import fetch from 'cross-fetch';
+
+export const author = 'perror';
+export const version = '1.0.0';
+
+const abi = [
+  'function balanceOfEVO(address _account, uint start, uint end) public view returns (uint256 lands, uint apostles)',
+  'function totalOfEVO(address _account) public view returns (uint total)',
+  'function balanceOfStaking(address _account) external view returns (uint256)'
+];
+
+let row = 750;
+
+const get_heco_account_vote = async (addresses, api, block, network) => {
+  const resp = await fetch(api + '/api/snapshot/vote/', {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({ addresses: addresses, block: block, chain: network })
+  });
+  return await resp.json();
+};
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  if (options.row !== undefined && options.row !== 0 && options.row !== '') {
+    row = options.row;
+  }
+
+  const strategy = options;
+  if (strategy.api_host !== '' && strategy.api_host !== undefined) {
+    // get Heco network vote from evo backend
+    const resp = await get_heco_account_vote(
+      addresses,
+      strategy.api_host,
+      blockTag,
+      network
+    );
+    addresses.forEach((address) => {
+      if (!accountVote.hasOwnProperty(address)) {
+        accountVote[address] = 0;
+      }
+      if (resp.hasOwnProperty(address)) {
+        accountVote[address] += resp[address].land * strategy.land_multiplier;
+        accountVote[address] +=
+          resp[address].apostle * strategy.apostle_multiplier;
+      }
+    });
+    return;
+  }
+
+  const promiseArray: Promise<any>[] = [];
+  const promiseArrayInfo: string[] = [];
+  addresses.forEach((address) => {
+    promiseArrayInfo.push(address);
+    promiseArray.push(
+      call(getProvider(network), abi, [
+        strategy.address,
+        'totalOfEVO',
+        [address]
+      ])
+    );
+  });
+  const res = await Promise.all(promiseArray);
```

Comment:
Hi @PerrorOne Can you use multicall here plz, one call per one address like this won't scale for bigger  voting,

Here is an example of how to call multicall,
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/protofi-erc721-tier-weighted/index.ts#L28-L36

Reply:
Because the `multicall` query uses a block, which may cause the contract function to take too long and cause an 'execution reversed' error, so I used `call`. Is there a problem with this treatment?  thank you. @ChaituVR 

.

Reply:
Yes @PerrorOne Can't send one request per one address, https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Reply:
@PerrorOne